core/blockchain.go

```diff
@@ -2068,7 +2101,14 @@ func (bc *BlockChain) BuildAndWriteBlock(parentBlock *types.Block, header *types
 		l.BlockHash = blockHash
 	}
 
-	return bc.writeBlockAndSetHead(fullBlock, receipts, logs, statedb, false)
+	// Double check: even though we just built the block, make sure it is valid.
+	if err = bc.validator.ValidateState(fullBlock, statedb, receipts, gasUsed); err != nil {
+		bc.reportBlock(fullBlock, receipts, err)
+		return nil, NonStatTy, fmt.Errorf("error validating block %d: %w", fullBlock.Number().Uint64(), err)
+	}
+
+	writeStatus, err := bc.writeBlockAndSetHead(fullBlock, receipts, logs, statedb, false)
```

Comment:
Would set `emitHeadEvent` as true help clear the txn pool?

Reply:
Since this method is only used in L1 follower node I don't think it is necessary as all networking is disabled anyway

Reply:
oh OK. I thought it would help txn pool to reorg once the node generates new blocks. like here: https://github.com/scroll-tech/go-ethereum/blob/6f14c337e5fa7358a51c69195600f4ff3db949e6/core/txpool/txpool.go#L217.  and yeah I think it's not the major considerations during recovery, since the txns will be handled later too.